test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -294,6 +297,99 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 sendMalformedRequest: true);
         }
 
+        [Fact]
+        public async Task OnCompletedExceptionShouldNotPreventAResponse()
+        {
+            var hostBuilder = TransportSelector.GetWebHostBuilder()
+                .UseKestrel()
+                .UseUrls("http://127.0.0.1:0/")
+                .Configure(app =>
+                {
+                    app.Run(async context =>
+                    {
+                        context.Response.OnCompleted(_ => throw new Exception(), null);
+                        await context.Response.WriteAsync("hello, world");
+                    });
+                });
+
+            using (var host = hostBuilder.Build())
+            {
+                host.Start();
+
+                using (var client = new HttpClient())
+                {
+                    var response = await client.GetAsync($"http://127.0.0.1:{host.GetPort()}/");
+                    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                }
+            }
+        }
+
+        [Fact]
+        public async Task OnCompletedShouldNotBlockAResponse()
+        {
+            var delay = Task.Delay(TestConstants.DefaultTimeout);
```

Comment:
Do you really want this delay to star this early?

Reply:
I think it's OK. It makes it easy to access the `delay` Task from server and client parts of the test. Unless we're worried about this test ever taking over 30 seconds.